processing/src/main/java/org/apache/druid/data/input/InputSource.java

```diff
@@ -87,4 +88,10 @@ InputSourceReader reader(
       @Nullable InputFormat inputFormat,
       File temporaryDirectory
   );
+
+  @Nullable
+  default Set<String> getTypes()
+  {
+    return null;
+  }
```

Comment:
It's better here to return a `Set<Resource>` rather than `Set<String>`. It makes it clearer that this is used for security purposes, since `Resource` is a security-specific thing.

Reply:
`ResourceAction` isn't available to InputSources at the moment. I can add a dependency. Ok to do this? Not sure if it will create a dependency cycle

Reply:
I tried this and unfortunately it added a cyclic dependency